cobalt/web/agent.cc

```diff
@@ -577,19 +578,20 @@ void Agent::Run(const Options& options, InitializeCallback initialize_callback,
   if (!thread_.StartWithOptions(thread_options)) return;
   DCHECK(message_loop());
 
-  watchdog::Watchdog* watchdog = watchdog::Watchdog::GetInstance();
-
   // Registers service worker thread as a watchdog client.
-  if (watchdog) {
-    watchdog_name_ =
-        thread_.thread_name() + std::to_string(thread_.GetThreadId());
-    watchdog_registered_ = true;
-    watchdog->Register(watchdog_name_, watchdog_name_,
-                       base::kApplicationStateStarted, kWatchdogTimeInterval,
-                       kWatchdogTimeWait, watchdog::PING);
-    message_loop()->task_runner()->PostDelayedTask(
-        FROM_HERE, base::Bind(&Agent::PingWatchdog, base::Unretained(this)),
-        base::TimeDelta::FromMilliseconds(kWatchdogTimePing));
+  if (thread_.thread_name() == worker::ServiceWorkerConsts::kServiceWorker) {
+    watchdog::Watchdog* watchdog = watchdog::Watchdog::GetInstance();
+
+    if (watchdog) {
+      watchdog_name_ = worker::ServiceWorkerConsts::kServiceWorker;
```

Comment:
Won't this fail if there is more than one service worker? e.g. there may be a service worker installed and somehow left behind on a device from a service worker integration test, in addition to one for production use.

Reply:
What is the "service worker integration test" you mean here? As far as I know, we don't have watchdog instance in tests today. (I could be wrong, but @briantting should know the answer here.) 
Even if we have watchdog for testing, will we have two active SWs at the same time this case?

Reply:
There is no guarantee that there will be only one service worker. In fact when a service worker is being updated there will be two service workers until the new one becomes the controlling service worker.

Additionally there can be different service workers for different origins and scopes, and they can all be active at the same time.

Reply:
If there are multiple service workers using the same watchdog_name... behavior is going to get wonky. It'll take the latest registered one (depending on register arguments) and monitor both service workers as a single client. So that single client will be pinged by both service workers. This could be fine if you want to monitor all service workers as a whole single client. Alternatively you could have unique names for each one but then fetching violations from all of the unique names is trickier.

Perhaps grouping would need to be introduced.

Reply:
We should treat them as different clients. Suppose one SW is alive, it will constantly ping watchdog, then we won't detect if there is any other SW dead. I'm just going to keep everything the same and have thread_id in the name now.